### ISSUE TITLE
[develop2] MacOS arm64 architecture detection

### ISSUE THREAD

**Issue by fklebert** [conan-io/cmake-conan#537]
```
function(detect_arch ARCH)
    if(CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64)
        set(_ARCH armv8)
```
does not correctly detect MacOS arm64 (M1) as CMAKE_SYSTEM_PROCESSOR is set to `arm64`.
Changing it to 
```
if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|arm64")
        set(_ARCH armv8)
```
fixes it.

**Comment by jcar87** [comment]
Hi @fklebert  - thank you for reporting this.

We have a PR open that will include this fix: https://github.com/conan-io/cmake-conan/pull/529 - will look into merging this today.

**Comment by memsharded** [comment]
Closed by #529
